tests/integration/thrift_tests.py

```diff
@@ -406,6 +434,46 @@ def example(self, context):
 
         assert handler.edge_context == FakeEdgeContextFactory.DECODED_CONTEXT
 
+    def test_budget_header_pool_timeout(self):
+        """Test that the budget header is set in the headers if the client sets it."""
+        budget = 100.0
+
+        class Handler(TestService.Iface):
+            def example(self, context):
+                self.context = context
+                return True
+
+        handler = Handler()
+
+        span_observer = mock.Mock(spec=SpanObserver)
+        with serve_thrift(handler, TestService) as server:
+            with baseplate_thrift_client(server.endpoint, TestService, span_observer) as context:
+                with context.example_service.retrying(attempts=3, budget=budget) as svc:
+                    svc.example()
+
+        assert handler.context.headers.get(b"Deadline-Budget").decode() == str(1000)
+
+    def test_budget_header_retry_timeout(self):
+        """Test that the budget header is set in the headers if the client sets it."""
```

Comment:
this test actually tests when retry timeout is smaller than the pool timeout we pick the retry one correctly.

we need another test that when retry timeout is larger than the pool timeout we pick the pool one correctly. we probably need another one that only set the pool timeout and not the retry timeout.

also for retry timeout there can be the one from exponential backoff and the one from budget, we probably need separated tests for them as well.